src/services/token-list-service.ts

```diff
@@ -22,91 +23,69 @@ export class TokenListService {
     Registration.singleton(ITokenListService, TokenListService).register(container);
   }
 
-  /**
-   * address key is set to lowercase
-   */
-  public tokenInfos: Map<TokenAddressId, ITokenInfoUniswap> | undefined;
-
-  /**
-   * Hydrate this.tokenInfos from all configured TokenInfo documents for the current network.
-   */
   @callOnce('TokenListService')
   public async initialize(): Promise<void> {
-    if (typeof this.tokenInfos === 'undefined') {
-      this.timingService.startTimer('fetch tokeninfos');
-      if (!this.ethereumService.targetedNetwork) {
-        throw new Error(`TokenListService: no tokenlists available for the empty network`);
-      }
-      const uris = TokenLists.get(this.ethereumService.targetedNetwork);
-      if (!uris) {
-        throw new Error(`TokenListProvider: no tokenlists available for the network ${this.ethereumService.targetedNetwork}`);
-      }
-      /**
-       * get an array of promised ITokenInfoUniswap arrays, excluding any that are null or empty
-       */
-      const tokenInfoArrays = (await Promise.all(uris.map((uri) => this.fetch(uri)))).filter(
-        (tokenInfoArray) => tokenInfoArray?.length,
-      ) as ITokenInfoUniswap[][];
-      /**
-       * compile lists into a single map, filtering the tokeninfos by targetted network.
-       * Address key is set to lowercase.
-       */
-      const tokenInfoMap = new Map<Address, ITokenInfoUniswap>();
-      for (const tokenInfoArray of tokenInfoArrays) {
-        for (const tokenInfo of tokenInfoArray) {
-          if (tokenInfo.chainId === this.ethereumService.targetedChainId) {
-            tokenInfoMap.set(tokenInfo.address.toLowerCase(), tokenInfo);
-          }
-        }
-      }
-      // if (tokenInfoMap.size === 0) {
-      //   throw new Error('Failed to load any TokenInfos');
-      // }
-      this.timingService.endTimer('fetch tokeninfos');
-      this.tokenInfos = tokenInfoMap;
+    if (this.tokenInfos) return;
+
+    this.timingService.startTimer('fetch tokeninfos');
+    const uris = this.ethereumService.targetedNetwork && TokenLists.get(this.ethereumService.targetedNetwork);
+
+    if (!uris) {
+      this.logger.fatal(`TokenListProvider: no tokenlists available for the network ${this.ethereumService.targetedNetwork ?? ''}`);
+      return;
```

Comment:
should end the timer before returning.

In general I am in favor of returning from functions only from a single location.